app/services/cache/MongoCacheClient.scala

```diff
@@ -192,17 +226,31 @@ class MongoCacheClient(appConfig: AppConfig, db: () => DefaultDB, applicationCry
     case c => c
   }
 
+  def fetchAll(id: String, deprecatedFilter: Boolean): Future[Option[Cache]] = {
+    val key = if(deprecatedFilter) bsonIdQueryOid(id) else bsonIdQuery(id)
+
+    collection.find(key).one[Cache] map {
+      case Some(c) if appConfig.mongoEncryptionEnabled => Some(new CryptoCache(c, compositeSymmetricCrypto))
+      case c => c
+    }
+  }
+
   /**
     * Fetches the whole cache and returns default where not exists
     */
-  def fetchAllWithDefault(id: String): Future[Cache] = fetchAll(id).map {
+  def fetchAllWithDefault(id: String, deprecatedFilter: Boolean): Future[Cache] =
```

Comment:
Is deprecated filter the best name?

Reply:
I could call it donut if you like :) any suggestions?